### ISSUE TITLE
Mystery sub=-5 for return_latent GPT logits

### ISSUE THREAD

**Issue by patrickltobing** [coqui-ai/TTS#3584]
### Describe the bug

In [gpt.py line 507](https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/layers/xtts/gpt.py#L507), there is a mystery `sub=-5`, which the code authors themselves do not understand.

Apparently, it is because there is an addition of 3 to the `codes_lengths` in [line 402](https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/layers/xtts/gpt.py#L402), and they are used in [line 424-427](https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/layers/xtts/gpt.py#L424-L427) to check whether the provided `audio_codes` are shorter than the `max(codes_lengths)`, which will always be the case because you added those lengths with 3 beforehand, and then pad them to match the extended `codes_lengths + 3`.

Then, there is the addition of a `stop_token` at [line 441](https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/layers/xtts/gpt.py#L441), and finally the addition of a `start/stop token` at [line 453](https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/layers/xtts/gpt.py#L453). Hence, the total reduction of the length for GPT logits return in case of using `return_latent=True` will be 5, i.e., `sub=-5` as in [line 507](https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/layers/xtts/gpt.py#L507).

If the `audio_codes` is only appended with `start/stop token` as in line 453, the sub should be -1. If it also includes the line 441, the addition of a `stop token`, it is -2. Finally, by also including the addition of 3 paddings as the aforementioned, it is become -5.

### To Reproduce

Not in the context of reproducing bug.

### Environment

```shell
Not applicable.
```
